tests/parser/tests/cases/should_pass/pattern_matching_let/case.hash

```diff
@@ -0,0 +1,6 @@
+// struct desturcturing
+let Dog { name } = dog;
+let Dog { name = better_name } = dog;
+
+// namespace destructuring
+let { name: even_better_name} = dog;
```

Comment:
What do you think about having "=" for namespace destructuring, same as structs?

Reply:
I think it would be a lot better to standardise syntax and make the parsing code a little cleaner.

Reply:
Yeah I agree.

Reply:
Ill make an issue after this is merged.